src/OrchardCore/OrchardCore/Shell/ShellHost.cs

```diff
@@ -29,7 +29,8 @@ public class ShellHost : IShellHost, IDisposable
         private readonly ILogger _logger;
 
         private bool _initialized;
-        private ConcurrentDictionary<string, ShellContext> _shellContexts = new ConcurrentDictionary<string, ShellContext>();
+        private readonly ConcurrentDictionary<string, ShellContext> _shellContexts = new ConcurrentDictionary<string, ShellContext>();
+        private readonly ConcurrentDictionary<string, ShellSettings> _shellSettings = new ConcurrentDictionary<string, ShellSettings>();
```

Comment:
I don't see anything removing shellsettings from this dictionary?

Did I miss something or are they accumulating / being overwritten on line 187

Reply:
@deanmarcussen 

When a given shell is created, it is added to the list and **never removed permanently** as we can't remove a shell at runtime. **It can be removed temporarily** when released / reloaded, but just before a new shell is re-added.

The new shell settings collection is updated at the 3 places where we call `_shellContexts.TryAdd()` and if it returns true, **but a given shell settings is never removed, only updated**, so that we can safely use `GetSettings()`.

Reply:
Cool, thanks for clarifiying @jtkech